virtual-desktop/src/app/plugin-manager/plugin-manager.module.ts

```diff
@@ -10,13 +10,66 @@
   Copyright Contributors to the Zowe Project.
 */
 
-import { NgModule } from '@angular/core';
+import { NgModule, APP_INITIALIZER, LOCALE_ID  } from '@angular/core';
+import { registerLocaleData } from '@angular/common';
 
 import { PluginManager } from './shared/plugin-manager';
 import { PluginLoader } from './shared/plugin-loader';
 import { Angular2PluginFactory } from './plugin-factory/angular2/angular2-plugin-factory';
 import { IFramePluginFactory } from './plugin-factory/iframe/iframe-plugin-factory';
 import { ReactPluginFactory } from './plugin-factory/react/react-plugin-factory';
+import { Globalization } from '../shared/globalization';
+import { LanguageLocaleService } from '../shared/language-locale.service';
+
+const logger: ZLUX.ComponentLogger = ZoweZLUX.logger.makeComponentLogger('org.zowe.zlux.virtual-desktop.plugin-manager');
+
+export function localeIdFactory(localeService: LanguageLocaleService) {
+  const zoweGlobal = (window as any).ZoweZLUX;
+
+  // chicken and egg bootstrapping problem. virtual-desktop wants a particular implementation
+  // of globalization that can use cookies set by the browser-preferences service of this plugin
+  // We may move that service into zlux-proxy-server, then Globalization can maybe move to bootstrap.
+  if (!(zoweGlobal.globalization instanceof Globalization)) {
+    logger.info('Setting ZoweZLUX.globalization to an implementation specific to com.rs.mvd.ng2desktop')
```

Comment:
Note: we're upgrading a global and this is the first object that works in this way so we may need to review later on if we want to keep this, refine this, do more of this, etc.